radicle-cli/src/commands/patch/list.rs

```diff
@@ -16,17 +17,25 @@ pub fn run(
     repository: &Repository,
     profile: &Profile,
     workdir: Option<git::raw::Repository>,
+    filter: Option<patch::State>,
 ) -> anyhow::Result<()> {
     let me = *profile.id();
     let patches = Patches::open(repository)?;
-    let proposed = patches.proposed()?;
+    let all = patches.all()?;
 
     // Patches the user authored.
     let mut own = Vec::new();
     // Patches other users authored.
     let mut other = Vec::new();
 
-    for (id, patch, _) in proposed {
+    for patch in all {
+        let (id, patch, _) = patch?;
+
+        if let Some(filter) = filter {
+            if patch.state() != filter {
+                continue;
+            }
+        }
```

Comment:
Would this be a performant alternative?

```
        if filter == Some(patch.state()) {
                continue;
        }
```

Reply:
The first option could work, but we're calling `patch.state()` even if filter is `None`, so it seems wasteful.
`is_some_and` is not part of stable rust.

Reply:
I think you can use [`contains`](https://doc.rust-lang.org/std/option/enum.Option.html#method.contains) but unfortunately that doesn't give you the laziness either